src/cli/serve/integration_tests/reload_logging_config.test.js

```diff
@@ -110,8 +114,11 @@ describe('Server logging configuration', function () {
         isJson = false;
         setLoggingJson(false);
 
-        // reload logging config
-        child.kill('SIGHUP');
+        // Reload logging config. We give it a little bit of time to just make
+        // sure the process sighup handler is registered.
+        setTimeout(() => {
+          child.kill('SIGHUP');
+        }, 100);
```

Comment:
This is pretty ugly. (I mean that in the nicest way.) I know why we're doing this, but is there not a way based on callbacks that we could do this?

Reply:
I agree, but if it makes the test less flaky I'm fine with it. The difficult thing about spawning the process like we do in this test, is that you don't have any way of registering mocks etc, so you can only go by what's in stdout. I'm not 100% sure what's causing this yet, but my best guess right now is that we're `kill(...)`-ing before `process.on('SIGHUP'` is registered.